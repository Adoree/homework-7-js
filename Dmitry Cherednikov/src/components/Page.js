import React from 'react';
import PropTypes from 'prop-types';

const Page = ({
  name,
  id,
  date,
  onBack,
}) => {
  let elem;

  if (date) {
    elem = (
      <p>
        {date}
      </p>
    )
  } else {
    elem = (
      <p>
        Not yet catched
      </p>
    )
  }

  return (
    <div className="page">
      <img
        src={`https://raw.githubusercontent.com/epam-js-may-2018/homework-7-js/master/pokemons/${id}.png`}
        className="img img-large"
        alt={`${name}.png`}
      />
      <h1>
        {name}
      </h1>
      {elem}
      <a
        className=""
        onClick={(e) => {
          e.preventDefault();
          onBack();
        }}
      >
        Back
      </a>
    </div>
  )
};

Page.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  date: PropTypes.string,
  onBack: PropTypes.func.isRequired,
};

export default Page;