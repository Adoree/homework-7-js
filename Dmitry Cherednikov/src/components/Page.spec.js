import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Page from './Page';

const setup = () => {
  const props = {
    name: 'pidgey',
    id: 16,
    date: null,
    onBack: jest.fn(),
  };

  const output = shallow((
    <Page { ...props } />
  ));

  return {
    output,
    props,
  }
};

describe('<Page />', () => {
  it('should render correctly', () => {
    const { output } = setup();

    expect(toJson(output)).toMatchSnapshot();
  });

  it('renders message if its not catched', () => {
    const { output } = setup();
    const message = 'Not yet catched';

    expect(output.find('p')).toHaveLength(1);
    expect(output.find('p').text()).toEqual(message);
  });

  it('renders message with date if its catched', () => {
    const { output } = setup();
    const date = '08.06.2016, 17:46:24';

    output.setProps({ date });

    expect(output.find('p')).toHaveLength(1);
    expect(output.find('p').text()).toEqual(date);
  });

  it('successfully calls the onClick handler', () => {
    const { output, props } = setup();
    const preventDefault = () => {};

    output.find('a').simulate('click', { preventDefault });
    expect(props.onBack.mock.calls.length).toEqual(1);
  })
});