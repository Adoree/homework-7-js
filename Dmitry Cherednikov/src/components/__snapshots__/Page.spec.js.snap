// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> should render correctly 1`] = `
<div
  className="page"
>
  <img
    alt="pidgey.png"
    className="img img-large"
    src="https://raw.githubusercontent.com/epam-js-may-2018/homework-7-js/master/pokemons/16.png"
  />
  <h1>
    pidgey
  </h1>
  <p>
    Not yet catched
  </p>
  <a
    className=""
    onClick={[Function]}
  >
    Back
  </a>
</div>
`;
